Firmware/ChameleonMini/Application/MifareUltralight.c

```diff
@@ -484,13 +485,18 @@ uint16_t MifareUltralightAppProcess(uint8_t* Buffer, uint16_t BitCount)
             }
 
             return BitCount;
+        } else if (Cmd == CMD_READ && ByteCount == (CMD_READ_FRAME_SIZE + ISO14443A_CRCA_SIZE) && Buffer[1] == 0) {
+            /* This is a short activation method */
+            State = STATE_ACTIVE;
+            return AppProcess(Buffer, CMD_READ_FRAME_SIZE);
```

Comment:
As states READY1 and READY2 seem to have a quite common code base, couldn't they be factored to avoid code duplication and memory loading? Maybe like:
```
case STATE_READY1:
case STATE_READY2:
    // Variables declaration
    if( State == most probable state)
        // specific state variables init
    else if (State == least probable state)
       // specific state init
   // common code with set specific variables
   break;
```

Reply:
fixed